flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowEmitStrategy.scala

```diff
@@ -179,9 +178,10 @@ object WindowEmitStrategy {
 
   // It is a experimental config, will may be removed later.
   @Experimental
-  val TABLE_EXEC_EMIT_EARLY_FIRE_DELAY: ConfigOption[String] =
+  val TABLE_EXEC_EMIT_EARLY_FIRE_DELAY: ConfigOption[Duration] =
   key("table.exec.emit.early-fire.delay")
-      .noDefaultValue
+      .durationType()
+      .defaultValue(Duration.ofMillis(-1))
```

Comment:
It's not allowed to use `-1ms` for Duration type. We can use `.noDefaultValue()` to keep the same behavior as before. You can use `tableConfig.getConfiguration.getOptional(TABLE_EXEC_EMIT_EARLY_FIRE_DELAY).isPresent` to check whether it is set.